hack/generator/pkg/codegen/pipeline_stage.go

```diff
@@ -30,11 +32,33 @@ func MakePipelineStage(
 	return PipelineStage{
 		id:          id,
 		description: description,
-		Action:      action,
+		action:      action,
 	}
 }
 
 // HasId returns true if this stage has the specified id, false otherwise
 func (stage *PipelineStage) HasId(id string) bool {
 	return stage.id == id
 }
+
+// UsedFor specifies that this stage should be used for only the specified targets
+func (stage PipelineStage) UsedFor(targets ...PipelineTarget) PipelineStage {
+	stage.targets = targets
+	return stage
+}
+
+// IsTargetted returns true if this stage should only be used for specific targets
+func (stage *PipelineStage) IsTargetted() bool {
+	return len(stage.targets) > 0
+}
+
+// IsUsedFor returns true if this stage should be used for the specified target
+func (stage *PipelineStage) IsUsedFor(target PipelineTarget) bool {
```

Comment:
It might make more sense to just have this return true when the stage has no targets? Then you don't need IsTargeted above.